### PR TITLE
6/api/reviews/ id/comments

### PR DESCRIPTION
**Pull request by JakeMaciver**

### CODE REVIEW

File: __tests__/endpoint.test.js [hunk 1]
```diff
@@ -139,3 +139,62 @@ describe('GET /api/reviews', () => {
...
+				expect(comments).toHaveLength(3);
+				comments.forEach((comment) => {
+					expect(comment).toEqual({
+						comment_id: expect.any(Number),
+						votes: expect.any(Number),
+						created_at: expect.any(String),
+						author: expect.any(String),
+						body: expect.any(String),
+						review_id: 2,
+					});
```
Comment (HeatherEver) [inline __tests__/endpoint.test.js:170]: i like that you've explicitly checked this value 👌

File: __tests__/endpoint.test.js [hunk 2]
```diff
@@ -139,3 +139,62 @@ describe('GET /api/reviews', () => {
...
+			});
+	});
+	test('200: should return 200 and an empty array when review_id exists but no comments', () => {
+		return request(app)
+			.get('/api/reviews/6/comments')
+			.expect(200)
+			.then(({ body }) => {
+				const { comments } = body;
+				expect(comments).toEqual([]);
+			});
```
Comment (HeatherEver) [inline __tests__/endpoint.test.js:190]: well done for adding in this test 👏

File: __tests__/endpoint.test.js [hunk 3]
```diff
@@ -139,3 +139,62 @@ describe('GET /api/reviews', () => {
...
+	});
+	test('404: should return a 404 not found error when review_id does not exist', () => {
+		return request(app)
+			.get('/api/reviews/999/comments')
+			.expect(404)
+			.then(({ body }) => {
+				expect(body).toEqual({ message: 'Not found' });
+			});
+	});
```
Comment (HeatherEver) [inline __tests__/endpoint.test.js:199]: can you please add just one more test 
`400 - /api/reviews/hello/comments`

File: __tests__/utils.test.js [hunk 4]
```diff
@@ -102,3 +103,22 @@ describe("formatComments", () => {
...
+			expect(typeof result).toBe('boolean');
+		});
+	});
+	test('404: returns a 404 error not found when value does not exist in column in the table', () => {
+		const table = 'reviews';
+		const column = 'review_id';
+		const value = 34;
+		return checkExists(table, column, value).then((result) => {
+			expect(result).toBe(false);
+		});
```
Comment (HeatherEver) [inline __tests__/utils.test.js:122]: well done for testing your utils function 👏

**Review by HeatherEver — APPROVED** [review]
Really nice work jake, i love that you've tested your utils function - really well done! 
Just one more test to add for this endpoint 
One more thing i would say, i notice you've used `async.... await` in your utils and comments functions this is great but i would stick to either `async` or `promises` in the same project for consistency purposes

File: __tests__/endpoint.test.js (hunk 3 above)
Comment (JakeMaciver) [reply __tests__/endpoint.test.js:199]: of course!

**Pull request merged**